marklogic-data-hub-central/ui/src/components/load/load-card.test.tsx

```diff
@@ -1,10 +1,118 @@
 import React from 'react';
-import { shallow } from 'enzyme';
+import { render, fireEvent, wait, cleanup, waitForElement } from '@testing-library/react';
+import { MemoryRouter } from 'react-router-dom';
 import LoadCard from './load-card';
+import data from '../../config/test-data.config';
+import axiosMock from 'axios';
+import mocks from '../../config/mocks.config';
+import { AuthoritiesService, AuthoritiesContext } from '../../util/authorities';
+
+jest.mock('axios');
+
+const mockHistoryPush = jest.fn();
+
+jest.mock('react-router-dom', () => ({
+  ...jest.requireActual('react-router-dom'),
+  useHistory: () => ({
+    push: mockHistoryPush,
+  }),
+}));
 
 describe('Load Card component', () => {
-  it('should render correctly', () => {
-    shallow(<LoadCard data deleteLoadArtifact
-      createLoadArtifact/>);
+
+  beforeEach(() => {
+    mocks.loadAPI(axiosMock);
+  })
+
+  afterEach(() => {
+    jest.clearAllMocks();
+    cleanup();
+  })
+
+  test('Load Card - Add step to an existing Flow', async () => {
+    const authorityService = new AuthoritiesService();
+    authorityService.setAuthorities(['readIngestion', 'writeIngestion', 'writeFlow']);
+    const { getByText, getByLabelText, getByTestId } = render(
+      <MemoryRouter>
+        <AuthoritiesContext.Provider value={authorityService}>
+          <LoadCard 
+            {...data.loadData}
+            flows={data.flows}
+            canWriteFlow={true}
+            addStepToFlow={jest.fn()}
+            addStepToNew={jest.fn()} />
+        </AuthoritiesContext.Provider>
+      </MemoryRouter>
+    )
+
+    //Check if the card is rendered properly
+    expect(getByText('Add New')).toBeInTheDocument();
+    expect(getByText('testLoadXML')).toBeInTheDocument();
+    expect(getByLabelText('testLoadXML-sourceFormat')).toBeInTheDocument();
+    expect(getByText('Last Updated: 04/15/2020 2:22PM')).toBeInTheDocument();
+    expect(await (waitForElement(() => getByLabelText('testLoadXML-stepName')))).toBeInTheDocument();
```

Comment:
any particular reason we are waiting for element here, there is no preceding fireEvent.

Reply:
It was added earlier but i guess i never removed it. Did it now as its not needed anymore.